master/internal/rm/tasklist/job.go

```diff
@@ -287,7 +288,15 @@ func SortTasksWithPosition(
 ) []*sproto.AllocateRequest {
 	var reqs []*sproto.AllocateRequest
 	for it := taskList.Iterator(); it.Next(); {
-		reqs = append(reqs, it.Value())
+		req := it.Value()
+		if groups[req.Group] == nil {
+			log.Errorf(
+				`found an allocation (%s) without a group (%s) when trying to sort by priority; ignoring it`,
+				req.Name, req.Group.Address(),
```

Comment:
in the spirit of trusting nothing

could req.Group be nil?

Reply:
sure, i'll validate it in `AllocateRequest`'s receive in a follow up.